api/v1/models/Location.js

```diff
@@ -0,0 +1,70 @@
+/* jshint esversion 6 */
+var _Promise = require('bluebird');
+var _ = require('lodash');
+
+var errors = require('../errors');
+
+var Model = require('./Model');
+var LocationEvents = require('./LocationEvents');
+
+var Location = Model.extend({
+        tableName: 'locations',
+        idAttribute: 'id',
+        validations: {
+                name: ['required', 'string', 'maxLength:255'],
+                shortName: ['required', 'string', 'maxLength:25'],
+                latitude: ['required', 'number'],
+                longitude: ['required', 'number']
+        },
+
+        locationEvents: function () {
+                return this.belongsToMany(LocationEvents);
+        }
+});
+
+/**
+ * Adds locations to the specified event. If the role already exists, it is returned
+ * unmodified
+ * @param  {id}          id          Event id. If undefined, the event will be created
+ * @param  {Event}       event       Event Object to add the event to
+ * @param  {[Location]}  locations   Array of Location objects. Containing three parameters:
+ *                                            name:      name of the location,
+ *                                            latitude:  latitude of the location,
+ *                                            longitude: longitude of the location,
+ * @throws NotFoundError             When and id that is not found is requested.
+ * @returns {Promise<Location>} the result of the addititon
+ */
+Location.addLocation = function(id, name, short_name, latitude, longitude) {
```

Comment:
Docstrings don't seem to match up to function signature. You can probably pass in an object of attributes here as well